terraso_backend/apps/graphql/schema/data_entries.py

```diff
@@ -48,7 +50,46 @@ def get_queryset(cls, queryset, info):
         return queryset.filter(groups__in=user_groups_ids)
 
     def resolve_url(self, info):
-        return self.signed_url
+        if (self.entry_type == DataEntry.ENTRY_TYPE_FILE):
+            return self.signed_url
+        return self.url
+
+
+class DataEntryAddMutation(BaseWriteMutation):
+    data_entry = graphene.Field(DataEntryNode)
+
+    model_class = DataEntry
+
+    class Input:
+        group_slug = graphene.String(required=True)
+        name = graphene.String(required=True)
+        url = graphene.String(required=True)
+        entry_type = graphene.String(required=True)
+        resource_type = graphene.String(required=True)
+        description = graphene.String()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+
+        if not cls.is_update(kwargs):
+            kwargs["created_by"] = user
+
+        if "entry_type" in kwargs:
+            kwargs["entry_type"] = DataEntry.get_entry_type_from_text(kwargs["entry_type"])
+
+        group_slug = kwargs.pop("group_slug")
+        try:
+            group = Group.objects.get(slug=group_slug)
+        except Group.DoesNotExist:
+            logger.error("Group not found when adding Data Entry", extra={"group_slug": group_slug})
```

Comment:
question: do we want to use `logger.exception`?

Reply:
@paulschreiber I don't think so, it seems that that log will give us all the information that we need, I don't think we need the stack trace, or what do you think? should I change it?

Reply:
@david-code what do you think?

Reply:
I think it's fine just to log the message here. Because the exception caught is `Group.DoesNotExist`, the stack trace will just let us know that the group doesn't exist and shouldn't have any extra information.